docs/backend/schema.ddl

```diff
@@ -0,0 +1,281 @@
+///////////////////////////////////////////////////
+// Groups and users
+
+Table Group {
+  id INT [pk, increment]
+  name VARCHAR [not null, unique]
+  gisaid_division_name VARCHAR [unique]
+  email VARCHAR [not null, unique]
+}
+
+Table User {
+  id INT [pk, increment]
+  name VARCHAR [not null, unique]
+  email VARCHAR [not null, unique]
+  auth0_user_id VARCHAR [not null]
+  group INT [not null, ref: > Group.id]
+  group_admin bool [not null]
+  system_admin bool [not null]
+}
+
+
+///////////////////////////////////////////////////
+// Cross-group viewing
+
+Table TypeOfData {
+  // Types of data people can view.  This table should be pre-populated with at least the following entries:
+  //  1. Trees
+  //  2. Sequences
+  id INT [pk, increment]
+  name VARCHAR [not null]
+}
+
+// members of group A (viewer_group) can see data uploaded by group B (owner_group)
+Table GroupCanSee {
+  viewer_group_id INT [not null, ref: > Group.id]
+  owner_group_id INT [not null, ref: > Group.id]
+  type_of_data_id INT [not null, ref: > TypeOfData.id]
+
+  Indexes {
+    (viewer_group_id, owner_group_id, type_of_data_id) [unique]
+  }
+}
+
+
+///////////////////////////////////////////////////
+// Physical Samples
+
+Table PhysicalSample {
```

Comment:
This needs a column for the `isolate` (the name used for submission to GISAID/Genbank/etc)

Reply:
We should also add a column for `purpose of sampling` as recommended by PH4AGE. Important to distinguish between samples collected for surveillance, for outbreak investigation, and for targeted searches e.g. S-gene dropout

Reply:
`isolate` would all be SARS-CoV-2, right?  this is futureproofing, I assume?

Reply:
No, `isolate` is what we've been calling `gisaid_name` -- basically, it's the human-readable public name of the object, and is usually something like `USA/CA-CZB-1234/2021`